api/python/t4/api.py

```diff
@@ -208,27 +210,183 @@ def list_packages(registry=None):
     Returns:
         A list of strings containing the names of the packages
     """
-    registry = get_package_registry(fix_url(registry) if registry else None) + '/named_packages'
+    class PackageList:
+        """Display wrapper for list_packages"""
+
+        def __init__(self, pkg_info):
+            self.pkg_names = [info['pkg_name'].replace(':latest', '') for info in pkg_info]
+            self._repr = self.create_str(pkg_info)
+
+        def __repr__(self):
+            return self._repr
+
+        def __iter__(self):
+            return iter(self.pkg_names)
 
-    registry_url = urlparse(registry)
+        def __len__(self):
+            return len(self.pkg_names)
+
+        def __contains__(self, item):
+            return item in self.pkg_names
+
+        @staticmethod
+        def _fmt_str(string, strlen):
+            """Formats strings to a certain width"""
+            if len(string) > strlen - 3:
```

Comment:
should this check be `> strlen -3` or just `> strlen` ?

Reply:
Actually it should be `len(string) > strlen - 1`, since the old spacer was three characters wide (`...`) and the new spacer is just one character wide (`…`).